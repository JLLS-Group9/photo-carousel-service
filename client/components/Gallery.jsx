import React from 'react';
import styles from '../style.css';

const Gallery = ({listingInfo}) => {

  if (listingInfo.photos) {
    return (
      <div>
        <div>

        </div>
        <div>{listingInfo.address} | ${listingInfo.price.toLocaleString()} | {listingInfo.bedroom} Beds {listingInfo.bathroom} Baths</div>
      </div>
    );
  } else {
    return null;
  }


};

export default Gallery;