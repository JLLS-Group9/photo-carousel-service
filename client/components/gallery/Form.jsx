import React from 'react';
import styles from './gallery.css';

const Form = (props) => {
  return (
    <div className={styles.form}>
      <div className={styles.content}>
        <div className={styles.formHeader}>Contact This Property</div>
        <div>
          <form>
            <input type="text" id="fname" name="name" placeholder="Your Name" />
            <input type="text" id="fname" name="number" placeholder="Your Phone Number" />
            <input type="text" id="fname" name="email" placeholder="Your Email"/>
            <textarea name="tutorial" rows="3" cols="40" placeholder="I am interested in this rental and would like to schedule a viewing. Please let me know when this would be possible."></textarea>
            <input className={styles.submit} type="submit" value="Check Availability"></input>
            <label className={styles.container}>
              <input type="checkbox" className={styles.checkmark}/>
              &ensp;Enable 1-Click Request
            </label>
          </form>
        </div>
        <div className={styles.warning}>You agree to Trulia's Terms of Use {'&'} Privacy Policy By choosing to contact a property, you also agree that Zillow Group, landlords, and property managers may call or text you about any inquiries you submit through our services, which may involve use of automated means and prerecorded/artificial voices. You don't need to consent as a condition of renting any property, or buying any other goods or services. Message/data rates may apply.</div>
      </div>
    </div>
  );
};

export default Form;