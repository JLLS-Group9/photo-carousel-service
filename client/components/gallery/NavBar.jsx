import React from 'react';
import styles from './gallery.css';

const NavBar = ({save, listingInfo, toggleGallery}) => (
  <div>
    <div className={styles.navBar}>
      <span className={styles.navborders, styles.navPhoto}>Photos</span>
      <span className={styles.navborders}>Map</span>
      <span className={styles.navborders}>Street View</span>
      <span className={styles.navborders}>Schools</span>
      <span className={styles.navborders}>Crime</span>
      <span className={styles.navborders}>Commute</span>
      <span className={styles.navborders}>Stop {'&'} Shop</span>
    </div>
    <div className={styles.gridButtons}>
      <span className={styles.save} onClick={()=> { save(); }}>
        <svg className={styles.heartsvg} width="25" height="25" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
          {listingInfo.saved ? <path d="M16.157 6.31A7.874 7.874 0 1 1 27.3 17.433l-1.913 1.912-9.254 9.254-1.88-1.88-7.373-7.374-1.91-1.91a7.874 7.874 0 1 1 11.137-11.13l.027.025.022-.022z" fill="rgb(255, 94, 63)"></path> : <path d="M26.95 11.863a5.214 5.214 0 0 0-8.908-3.677l-1.908 1.908-1.906-1.908a5.214 5.214 0 1 0-7.377 7.366l1.912 1.913 7.371 7.373 7.373-7.373 1.912-1.912a5.193 5.193 0 0 0 1.53-3.69zM16.157 6.31A7.874 7.874 0 1 1 27.3 17.433l-1.913 1.913-9.254 9.254-1.88-1.88-7.373-7.374-1.91-1.91a7.874 7.874 0 1 1 11.137-11.13l.027.025.022-.022z" fill="rgb(0, 120, 130)"></path>}

        </svg>
        Save
      </span>
      <span className={styles.share}>
        <svg className={styles.sharesvg} width="25" height="25" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg"><path d="M17.29 7.2v14.285h-2.66V7.201l-3.99 3.99L8.76 9.31l7.2-7.2 7.2 7.2-1.88 1.88-3.99-3.99zm5.32 9.298h-2.66v-2.66h5.32v15.295H6.65V13.838h5.32v2.66H9.31v9.975h13.3v-9.975z" fill="rgb(0, 120, 130)"></path></svg> Share</span>
      <span className={styles.closesvg} onClick={() => { toggleGallery(); } }>
        <svg width="24" height="24" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg"><path className={styles.cover} d="M27.816 25.935l-1.881 1.88-21.83-21.83 1.88-1.88 21.83 21.83zm-1.881-21.83l1.88 1.88-21.83 21.83-1.88-1.88 21.83-21.83z" fill="#869099"></path></svg></span>
    </div>
  </div>
);

export default NavBar;