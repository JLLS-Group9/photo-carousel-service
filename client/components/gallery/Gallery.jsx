import React from 'react';
import NavBar from './NavBar.jsx';
import PhotoGrid from './PhotoGrid.jsx';
import styles from './gallery.css';

const Gallery = ({listingInfo, show, toggleGallery, toggleCarousel}) => {


  const closeModal = () => {
    let modal = document.getElementById('myModal');
    if (event.target === modal) {
      toggleGallery();
    }
  };

  //if show is true modal is shown
  if (show) {

    return (
      <div>
        {/*This is wrapper for the entire modal*/ }
        {/*closeModal changes the state of showGallery from the function in line 9*/ }
        <div className={styles.Modal} onClick={closeModal} id="myModal">
           {/*This is wrapper for the modal content */ }
          <div className={styles.ModalContent}>
            <NavBar />
            <div className={styles.listingHeader}>{listingInfo.address} | ${listingInfo.price.toLocaleString()} | {listingInfo.bedroom} Beds {listingInfo.bathroom} Baths</div>
            <PhotoGrid photos={listingInfo.photos} toggleCarousel={toggleCarousel}/>
          </div>
        </div>
      </div>
    );

  } else {
    return null;
  }


};

export default Gallery;