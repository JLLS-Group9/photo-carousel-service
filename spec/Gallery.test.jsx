import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {mount, shallow} from 'enzyme';
import { spy } from 'sinon';
import Gallery from '../client/components/gallery/Gallery.jsx';
import NavBar from '../client/components/gallery/NavBar.jsx';
import Form from '../client/components/gallery/Form.jsx';
import data from './dummydata.js';

Enzyme.configure({ adapter: new Adapter() });


describe('<NavBar />', () => {
  xit('renders NavBar', () => {
    const wrapper = shallow(<NavBar save={()=>{}} toggleGallery={()=> {}} listingInfo={data}/>);
    expect(wrapper).toMatchSnapshot();
  });


  xit('renders Form', () => {
    const wrapper = shallow(<Form/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders NavBar', () => {
    const wrapper = shallow(<NavBar save={()=>{}} toggleGallery={()=> {}} listingInfo={data}/>);
    expect(wrapper).not.toBeEmptyRender();
  });

  it('renders Form', () => {
    const wrapper = shallow(<Form/>);
    expect(wrapper).not.toBeEmptyRender();
  });

  // it('has svg files', () => {
  //   const wrapper = shallow(<NavBar save={()=>{}} toggleGallery={()=> {}} listingInfo={data}/>);
  // });



  it('should not Gallery render when show is false', ()=> {
    const wrapper = shallow(<Gallery listingInfo={data} show={false} numPhoto={data.photos.length} toggleGallery={()=>{}} toggleCarousel={()=>{}} addPlace={()=>{}} save={()=>{}}/>);
    expect(wrapper).toBeEmptyRender();
  });

  it('should render Gallery when show is true', ()=> {
    const wrapper = shallow(<Gallery listingInfo={data} show={true} numPhoto={data.photos.length} toggleGallery={()=>{}} toggleCarousel={()=>{}} addPlace={()=>{}} save={()=>{}}/>);
    expect(wrapper).not.toBeEmptyRender();
  });

  it('trigger toggleGallery when clicked', ()=> {
    const wrapper = shallow(<Gallery listingInfo={data} show={true} numPhoto={data.photos.length} toggleGallery={spy()} toggleCarousel={spy()} addPlace={spy()} save={spy()}/>);

    wrapper
      .find('.save')
      .simulate('click');

    expect(spy.calledOnce).toBe(true);

  });

});


