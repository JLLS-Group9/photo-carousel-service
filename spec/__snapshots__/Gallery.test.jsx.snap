// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders 1`] = `
<div>
  <div
    className="navBar"
  >
    <span
      className="navPhoto"
    >
      Photos
    </span>
    <span
      className="navborders"
    >
      Map
    </span>
    <span
      className="navborders"
    >
      Street View
    </span>
    <span
      className="navborders"
    >
      Schools
    </span>
    <span
      className="navborders"
    >
      Crime
    </span>
    <span
      className="navborders"
    >
      Commute
    </span>
    <span
      className="navborders"
    >
      Stop 
      &
       Shop
    </span>
  </div>
  <div
    className="gridButtons"
  >
    <span
      className="save"
      onClick={[Function]}
    >
      <svg
        className="heartsvg"
        height="25"
        viewBox="0 0 32 32"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M26.95 11.863a5.214 5.214 0 0 0-8.908-3.677l-1.908 1.908-1.906-1.908a5.214 5.214 0 1 0-7.377 7.366l1.912 1.913 7.371 7.373 7.373-7.373 1.912-1.912a5.193 5.193 0 0 0 1.53-3.69zM16.157 6.31A7.874 7.874 0 1 1 27.3 17.433l-1.913 1.913-9.254 9.254-1.88-1.88-7.373-7.374-1.91-1.91a7.874 7.874 0 1 1 11.137-11.13l.027.025.022-.022z"
          fill="rgb(0, 120, 130)"
        />
      </svg>
      Save
    </span>
    <span
      className="share"
    >
      <svg
        className="sharesvg"
        height="25"
        viewBox="0 0 32 32"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.29 7.2v14.285h-2.66V7.201l-3.99 3.99L8.76 9.31l7.2-7.2 7.2 7.2-1.88 1.88-3.99-3.99zm5.32 9.298h-2.66v-2.66h5.32v15.295H6.65V13.838h5.32v2.66H9.31v9.975h13.3v-9.975z"
          fill="rgb(0, 120, 130)"
        />
      </svg>
       Share
    </span>
    <span
      className="closesvg"
      onClick={[Function]}
    >
      <svg
        height="24"
        viewBox="0 0 32 32"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="cover"
          d="M27.816 25.935l-1.881 1.88-21.83-21.83 1.88-1.88 21.83 21.83zm-1.881-21.83l1.88 1.88-21.83 21.83-1.88-1.88 21.83-21.83z"
          fill="#869099"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`<NavBar /> renders Form 1`] = `
<div
  className="form"
>
  <div
    className="content"
  >
    <div
      className="formHeader"
    >
      Contact This Property
    </div>
    <div>
      <form>
        <input
          id="fname"
          name="name"
          placeholder="Your Name"
          type="text"
        />
        <input
          id="fname"
          name="number"
          placeholder="Your Phone Number"
          type="text"
        />
        <input
          id="fname"
          name="email"
          placeholder="Your Email"
          type="text"
        />
        <textarea
          cols="40"
          name="tutorial"
          placeholder="I am interested in this rental and would like to schedule a viewing. Please let me know when this would be possible."
          rows="3"
        />
        <input
          className="submit"
          type="submit"
          value="Check Availability"
        />
        <label
          className="container"
        >
          <input
            className="checkmark"
            type="checkbox"
          />
           Enable 1-Click Request
        </label>
      </form>
    </div>
    <div
      className="warning"
    >
      You agree to Trulia's Terms of Use 
      &
       Privacy Policy By choosing to contact a property, you also agree that Zillow Group, landlords, and property managers may call or text you about any inquiries you submit through our services, which may involve use of automated means and prerecorded/artificial voices. You don't need to consent as a condition of renting any property, or buying any other goods or services. Message/data rates may apply.
    </div>
  </div>
</div>
`;

exports[`<NavBar /> renders NavBar 1`] = `
<div>
  <div
    className="navBar"
  >
    <span
      className="navPhoto"
    >
      Photos
    </span>
    <span
      className="navborders"
    >
      Map
    </span>
    <span
      className="navborders"
    >
      Street View
    </span>
    <span
      className="navborders"
    >
      Schools
    </span>
    <span
      className="navborders"
    >
      Crime
    </span>
    <span
      className="navborders"
    >
      Commute
    </span>
    <span
      className="navborders"
    >
      Stop 
      &
       Shop
    </span>
  </div>
  <div
    className="gridButtons"
  >
    <span
      className="save"
      onClick={[Function]}
    >
      <svg
        className="heartsvg"
        height="25"
        viewBox="0 0 32 32"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M26.95 11.863a5.214 5.214 0 0 0-8.908-3.677l-1.908 1.908-1.906-1.908a5.214 5.214 0 1 0-7.377 7.366l1.912 1.913 7.371 7.373 7.373-7.373 1.912-1.912a5.193 5.193 0 0 0 1.53-3.69zM16.157 6.31A7.874 7.874 0 1 1 27.3 17.433l-1.913 1.913-9.254 9.254-1.88-1.88-7.373-7.374-1.91-1.91a7.874 7.874 0 1 1 11.137-11.13l.027.025.022-.022z"
          fill="rgb(0, 120, 130)"
        />
      </svg>
      Save
    </span>
    <span
      className="share"
    >
      <svg
        className="sharesvg"
        height="25"
        viewBox="0 0 32 32"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.29 7.2v14.285h-2.66V7.201l-3.99 3.99L8.76 9.31l7.2-7.2 7.2 7.2-1.88 1.88-3.99-3.99zm5.32 9.298h-2.66v-2.66h5.32v15.295H6.65V13.838h5.32v2.66H9.31v9.975h13.3v-9.975z"
          fill="rgb(0, 120, 130)"
        />
      </svg>
       Share
    </span>
    <span
      className="closesvg"
      onClick={[Function]}
    >
      <svg
        height="24"
        viewBox="0 0 32 32"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="cover"
          d="M27.816 25.935l-1.881 1.88-21.83-21.83 1.88-1.88 21.83 21.83zm-1.881-21.83l1.88 1.88-21.83 21.83-1.88-1.88 21.83-21.83z"
          fill="#869099"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`<NavBar /> renders form 1`] = `
<div
  className="form"
>
  <div
    className="content"
  >
    <div
      className="formHeader"
    >
      Contact This Property
    </div>
    <div>
      <form>
        <input
          id="fname"
          name="name"
          placeholder="Your Name"
          type="text"
        />
        <input
          id="fname"
          name="number"
          placeholder="Your Phone Number"
          type="text"
        />
        <input
          id="fname"
          name="email"
          placeholder="Your Email"
          type="text"
        />
        <textarea
          cols="40"
          name="tutorial"
          placeholder="I am interested in this rental and would like to schedule a viewing. Please let me know when this would be possible."
          rows="3"
        />
        <input
          className="submit"
          type="submit"
          value="Check Availability"
        />
        <label
          className="container"
        >
          <input
            className="checkmark"
            type="checkbox"
          />
           Enable 1-Click Request
        </label>
      </form>
    </div>
    <div
      className="warning"
    >
      You agree to Trulia's Terms of Use 
      &
       Privacy Policy By choosing to contact a property, you also agree that Zillow Group, landlords, and property managers may call or text you about any inquiries you submit through our services, which may involve use of automated means and prerecorded/artificial voices. You don't need to consent as a condition of renting any property, or buying any other goods or services. Message/data rates may apply.
    </div>
  </div>
</div>
`;
